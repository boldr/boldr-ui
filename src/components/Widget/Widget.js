/* @flow */
import React from 'react';
import Card from 'react-md/lib/Cards/Card';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import CardActions from 'react-md/lib/Cards/CardActions';
import CardText from 'react-md/lib/Cards/CardText';

type Props = {
  name?: string,
  subtitle?: string,
};

const Widget = (props: Props) => {
  return (
    <div className="boldr-widget">
      <Card>
        <CardTitle title={ props.name } />
        <CardText>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Donec mattis pretium massa. Aliquam erat volutpat. Nulla facilisi.
          Donec vulputate interdum sollicitudin. Nunc lacinia auctor quam sed pellentesque.
          Aliquam dui mauris, mattis quis lacus id, pellentesque lobortis odio.
        </CardText>
      </Card>
    </div>
  );
};

export default Widget;
