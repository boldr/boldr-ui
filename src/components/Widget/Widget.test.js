import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Widget from './Widget';

describe('<Widget />', () => {
  const wrapper = shallow(<Widget name="widget" />);
  it('renders <Widget /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
