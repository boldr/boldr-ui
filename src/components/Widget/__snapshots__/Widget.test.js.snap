// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> renders snapshot 1`] = `
<div
  className="boldr-widget"
>
  <Card
    expanderIconChildren="keyboard_arrow_down"
    expanderIconClassName="material-icons"
    expanderTooltipPosition="left"
  >
    <CardTitle
      title="widget"
    />
    <CardText
      component="section"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec mattis pretium massa. Aliquam erat volutpat. Nulla facilisi. Donec vulputate interdum sollicitudin. Nunc lacinia auctor quam sed pellentesque. Aliquam dui mauris, mattis quis lacus id, pellentesque lobortis odio.
    </CardText>
  </Card>
</div>
`;
